### PR TITLE
Cadip / Adgs files ingestion prefect flows

### PR DESCRIPTION
**Pull request by agrosu1978**
This pull request implements user stories rspy-85 and rspy-115.

### CODE REVIEW

File: rs_workflows/common.py
Comment (Padeanu) [inline rs_workflows/common.py:1]: Either print / log but not mix.

**Review by Padeanu — APPROVED** [review]

File: rs_workflows/common.py [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+    Notes:
+        - This function constructs a station-specific endpoint URL by appending a path
+          based on the station type to the provided base URL.
+        - For "ADGS" stations, the endpoint path is "/adgs/aux/".
+        - For "CADIP" stations, the endpoint path is "/cadip/CADIP/cadu/".
+        - If an unsupported station type is provided, a RuntimeError is raised.
+
+    """
+    # url = http://127.0.0.1:8000
+    match station:
```
Comment (Padeanu) [inline rs_workflows/common.py:347]: maybe use globals from line :13 :14?

File: rs_workflows/common.py [hunk 2]
```diff
@@ -0,0 +1,451 @@
+"""Docstring will be here."""
```
Comment (jgaucher-cs) [inline rs_workflows/common.py:1]: Can you add a simple docstring e.g. "Utility functions implementation"

File: rs_workflows/common.py [hunk 3]
```diff
@@ -0,0 +1,451 @@
...
+        if (
+            response.ok
+            and "name" in eval_response.keys()
+            and filename == eval_response["name"]
+            and "status" in eval_response.keys()
+        ):
+            return EDownloadStatus(eval_response["status"])
+
+    except requests.exceptions.RequestException as e:
+        print(f"Request exception caught when calling for status endpoint: {e}")
```
Comment (jgaucher-cs) [inline rs_workflows/common.py:87]: I agree with Ovidiu, use logger instead of print

File: rs_workflows/common.py [hunk 4]
```diff
@@ -0,0 +1,451 @@
...
+    Attributes:
+        user (str): The user associated with the configuration.
+        url (str): The URL for the station.
+        station (str): The station identifier.
+        mission (str): The mission identifier.
+        tmp_download_path (str): The temporary download path.
+        s3_path (str): The S3 path for storing downloaded files.
+    """
+
+    def __init__(  # pylint: disable=too-many-arguments
```
Comment (jgaucher-cs) [inline rs_workflows/common.py:118]: Maybe here you could use a @dataclass instead, see: https://www.dataquest.io/blog/how-to-use-python-data-classes/

or as you did it for GetKeysFromS3Config

File: rs_workflows/example.py [hunk 5]
```diff
@@ -1,5 +1,7 @@
+"""Docstring here"""
```
Comment (jgaucher-cs) [inline rs_workflows/example.py:1]: Add a real description (even if it's only one line)

File: rs_workflows/common.py
Comment (agrosu1978) [reply rs_workflows/common.py:1]: Removed print

File: rs_workflows/common.py (hunk 1 above)
Comment (agrosu1978) [reply rs_workflows/common.py:347]: Agree

File: rs_workflows/common.py [hunk 6]
```diff
@@ -0,0 +1,451 @@
...
+                x.tolist() for x in [*np.array_split(files_stac, min(config.max_workers, len(files_stac)))]
+            ]
+            logger.debug(f"type(tasks_files_stac) = {type(tasks_files_stac)}")
+        except ValueError:
+            logger.warning("No task will be started, the requested number of tasks is 0 !")
+            tasks_files_stac = []
+
+        logger.info("List with files found in station = %s\n\n", files_stac)
+        idx = 0
+        for files_stac in tasks_files_stac:
```
[the remark this answers is not in the document]
Comment (agrosu1978) [reply rs_workflows/common.py:432]: Removed idx

File: rs_workflows/common.py (hunk 3 above)
Comment (agrosu1978) [reply rs_workflows/common.py]: Removed print

File: rs_workflows/common.py (hunk 4 above)
Comment (agrosu1978) [reply rs_workflows/common.py:119]: I tried with @dataclass in the beginning, but it seems somehow the prefect flow doesn't like it. For the task, there wasn't any problem, but the flow complained with an error (don't remember what error).

File: rs_workflows/example.py (hunk 5 above)
Comment (agrosu1978) [reply rs_workflows/example.py]: Done

**Comment by Padeanu** [comment]
Since only sonar is failing (due to some configuration problem) and all other quality checks pass, this PR can be merged.

**Pull request merged**